### PR TITLE
Syslog rewrite

### PR DESCRIPTION
**Pull request by tomneeman151293**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/28637

## Description
Syslog v2 rewrite from server integration.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details: Some of the filters for syslog that existed in V1 were not used. as can be seen in the issue above of the rewrite, it was decided to only implement message regex filter at first, and enhance with more filtering if requested.
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by tomneeman151293** [comment]
Validations are failing on:
- Missing TPB - the integration only implements long-running-execution so I didnt implement one
- using dev docker image - after this PR will be approved, i will merge my dockerfile PR https://github.com/demisto/dockerfiles/pull/6358, and those validations will pass

File: Packs/Syslog/Integrations/Syslogv2/Syslogv2.py [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+                prepare_globals_and_create_server(port, log_format, message_regex, incident_type, certificate,
+                                                  private_key)
+            except OSError as e:
+                if 'Address already in use' in str(e):
+                    raise DemistoException(f'Given port: {port} is already in use. Please either change port or '
+                                           f'make sure to close the connection in the server using that port.')
+                raise e
+            return_results('ok')
+        elif command == 'fetch-incidents':
+            fetch_samples()
```
Comment (dantavori) [inline Packs/Syslog/Integrations/Syslogv2/Syslogv2.py:316]: when exactly is this happening? let's document and briefly explain the sampling logic

File: Packs/Syslog/Integrations/Syslogv2/Syslogv2.py [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+def get_mapping_fields() -> Dict[str, str]:
+    def transfer_type_to_str_type(type_: Any):
+        type_as_str: str = str(type_)
+        # remove typing from the type name
+        type_as_str = type_as_str.replace('typing.', '')
+        # For classes other than typing, eg <class 'str'>
+        type_as_str = type_as_str.replace('<class ', '').replace("'", '').replace('>', '')
+        return type_as_str
+
+    return {k: transfer_type_to_str_type(v) for k, v in get_type_hints(SyslogMessageExtract).items()}
```
Comment (dantavori) [inline Packs/Syslog/Integrations/Syslogv2/Syslogv2.py:277]: nice impl, but better to provide the field description in the values of the dict, or which is used in which format. you can do it hard-coded.

**Review by dantavori — CHANGES_REQUESTED** [review]
🥇

File: Packs/Syslog/Integrations/Syslogv2/Syslogv2_description.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Configure Syslog in Cortex XSOAR
+- **Important**: Make sure to not use engine group for this integration. As it can cause the integration to run on different engine, meaning the Syslog server might send logs to an IP whom Syslog might not be configured at.
+- Configure a Syslog server to send it's logs to the Cortex XSOAR machine running the Syslog V2 integration.
+- Configure the Syslog format sent from Syslog server in Syslog v2 integration.
+- The integration **does not support** encrypted private key.
+- To create incidents from specific messages, use the Message Regex parameter to create incidents for logs who matches the given regex only
+- If no message regex is given. Incident will be created for each received Syslog message.
```
[the remark this answers is not in the document]
Comment (tomneeman151293) [reply Packs/Syslog/Integrations/Syslogv2/Syslogv2_description.md:7]: yeah, shouldn't be any issue, as the only problem is where the long-running needs to be configured on a single IP (in load balancing the IP might change so the server might not listen to the port in the expected IP

File: Packs/Syslog/Integrations/Syslogv2/Syslogv2.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  name: private_key
+  required: false
+  type: 14
+- display: Incoming Log Format
+  defaultvalue: Auto
+  hidden: false
+  name: log_format
+  options:
+  - RFC5424
+  - RFC3164
```
[the remark this answers is not in the document]
Comment (tomneeman151293) [reply Packs/Syslog/Integrations/Syslogv2/Syslogv2.yml:22]: https://stackoverflow.com/questions/69985784/syslog-rfc5424-vs-rfc6587

File: Packs/Syslog/Integrations/Syslogv2/Syslogv2.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  commands:
+  - name: get-mapping-fields
+    description: Returns the list of fields for an incident type. This command should
+      be used for debugging purposes.
+  isfetch: false
+  runonce: false
+  script: '-'
+  type: python
+  subtype: python3
+  dockerimage: devdemisto/syslog:1.0.0.24366
```
[the remark this answers is not in the document]
Comment (tomneeman151293) [reply Packs/Syslog/Integrations/Syslogv2/Syslogv2.yml:44]: leaving it as dev until dockerfiles pr is merged

File: Packs/Syslog/Integrations/Syslogv2/Syslogv2.py [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:
+    params = demisto.params()
+    command = demisto.command()
+    message_regex: Optional[str] = params.get('message_regex')
+    incident_type: Optional[str] = params.get('incident_type')
```
[the remark this answers is not in the document]
Comment (tomneeman151293) [reply Packs/Syslog/Integrations/Syslogv2/Syslogv2.py:287]: i will delete

File: Packs/Syslog/Integrations/Syslogv2/Syslogv2_description.md [hunk 7]
```diff
@@ -0,0 +1,8 @@
+## Configure Syslog in Cortex XSOAR
+- Configure a Syslog server to send its logs to the Cortex XSOAR machine that runs Syslog v2 integration.
+- Select the **Log Format** used by the Syslog server.
```
Comment (dantavori) [inline Packs/Syslog/Integrations/Syslogv2/Syslogv2_description.md:3]:
```suggestion
```

**Review by dantavori — COMMENTED** [review]
will approve after the demo.
try to think how we can test it in a TPB

**Comment by tomneeman151293** [comment]
@dantavori Updated docker image to prod image

**Review by dantavori — APPROVED** [review]
looks good 👍 please open an issue about the test playbook as we discussed

**Comment by tomneeman151293** [comment]
@dantavori  https://github.com/demisto/etc/issues/44245

**Comment by tomneeman151293** [comment]
@Itay4 need to force merge as we dont have a TPB for this integration.

**Comment by tomneeman151293** [comment]
@Itay4 Force merge is required. Validation is failing on missing TPB. A separate issue was created to find a dedicated instance to syslog TPB to be developed in the future

demisto/etc#44245

**Comment by Itay4** [comment]
merging due to the above

**Pull request merged**